src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java

```diff
@@ -0,0 +1,217 @@
+            .create();
+
+    /**
+     * Constructor to setup credentials provider
+     *
+     * @param cerberusUrl Cerberus URL
+     * @param region AWS Region used in auth with Cerberus
+     */
+    public StsCerberusCredentialsProvider(String cerberusUrl, String region) {
+        super(cerberusUrl);
+
+        if (region != null ) {
+            regionName = Regions.fromName(region).getName();
+        } else {
+            throw new CerberusClientException("Region is null. Please provide valid AWS region.");
+        }
+    }
+
+    /**
+     * Constructor to setup credentials provider
+     *
+     * @param cerberusUrl Cerberus URL
+     * @param region AWS Region used in auth with Cerberus
+     * @param xCerberusClientOverride Overrides the default header value for the 'X-Cerberus-Client' header
+     */
+    public StsCerberusCredentialsProvider(String cerberusUrl, String region, String xCerberusClientOverride) {
+        super(cerberusUrl, xCerberusClientOverride);
+        if (region != null ) {
+            regionName = Regions.fromName(region).getName();
+        } else {
+            throw new CerberusClientException("Region is null. Please provide valid AWS region.");
+        }
+    }
+
+    /**
+     * Constructor to setup credentials provider using the specified
+     * implementation of {@link OkHttpClient}
+     *
+     * @param cerberusUrl Cerberus URL
+     * @param region AWS Region used in auth with Cerberus
+     * @param httpClient the client for interacting with Cerberus
+     */
+    public StsCerberusCredentialsProvider(String cerberusUrl, String region, OkHttpClient httpClient) {
+        super(cerberusUrl, httpClient);
+        if (region != null ) {
+            regionName = Regions.fromName(region).getName();
+        } else {
+            throw new CerberusClientException("Region is null. Please provide valid AWS region.");
+        }
+    }
+
+    /**
+     * Obtains AWS Credentials.
+     */
+    private AWSCredentials getAWSCredentials(){
+        return DefaultAWSCredentialsProviderChain.getInstance().getCredentials();
+    }
+
+    /**
+     * Signs request using AWS V4 signing.
+     * @param request AWS STS request to sign
+     * @param credentials AWS credentials
+     */
+    private void signRequest(com.amazonaws.Request request, AWSCredentials credentials){
+
+        AWS4Signer signer = new AWS4Signer();
+        signer.setRegionName(regionName);
+        signer.setServiceName("sts");
+        signer.sign(request, credentials);
+    }
+
+    /**
+     * Generates and returns signed headers.
+     */
+    protected Map<String, String> getSignedHeaders(){
+
+        final String url = "https://sts." + regionName + ".amazonaws.com";
+
+        URI endpoint = null;
+
+        try {
+            endpoint = new URI(url);
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
```

Comment:
Never e.printStackTrace in production code, this should be log.error("Some message", e)

Reply:
fixed